sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java

```diff
@@ -88,108 +98,249 @@ public static void main(String[] args) throws IOException, InterruptedException
+                        .doOnTerminate(deleteRelationshipsSemaphore::release)
+                        .subscribe());
+            }
+
+            // Verify that the relationships have been deleted.
+            if (deleteRelationshipsSemaphore.tryAcquire(relationshipList.size(), MaxWaitTimeAsyncOperationsInSeconds, TimeUnit.SECONDS)) {
+                // Now the digital twin should be safe to delete
+
+                // Call APIs to delete the twins.
+                client.deleteDigitalTwin(twinId)
+                    .doOnSuccess(aVoid -> System.out.println("Deleted digital twin: " + twinId))
                     .doOnError(throwable -> {
-                        if (throwable instanceof ErrorResponseException && ((ErrorResponseException) throwable).getResponse().getStatusCode() == HttpStatus.SC_NOT_FOUND) {
-                            deleteRelationshipsSemaphore.release();
-                        } else {
-                            System.err.println("List incoming relationships error: " + throwable);
+                        if (!(throwable instanceof ErrorResponseException) || ((ErrorResponseException) throwable).getResponse().getStatusCode() != HttpStatus.SC_NOT_FOUND) {
+                            System.err.println("Delete twin error: " + throwable);
                         }
                     })
-                    .subscribe(
-                        incomingRelationship -> client.deleteRelationship(incomingRelationship.getSourceId(), incomingRelationship.getRelationshipId())
-                            .subscribe(
-                                aVoid -> System.out.println("Found and deleted incoming relationship: " + incomingRelationship.getRelationshipId()),
-                                throwable -> System.err.println("Delete incoming relationship error: " + throwable)
-                            ));
+                    .doOnTerminate(deleteTwinsLatch::countDown)
+                    .subscribe();
+
+                // Wait until the latch has been counted down for each async delete operation, signifying that the async call has completed successfully.
+                deleteTwinsLatch.await(MaxWaitTimeAsyncOperationsInSeconds, TimeUnit.SECONDS);
+            }
+        }
+    }
 
+    /**
+     * Delete models created by FullLifecycleSample for the ADT service instance.
+     * @throws InterruptedException If the current thread is interrupted while waiting to acquire latch.
+     */
+    public static void deleteAllModels() throws InterruptedException {
+        System.out.println("DELETING MODELS");
+
+        // This is to ensure models are deleted in an order such that no other models are referencing it.
+        List<String> models = asList(RoomModelId, WifiModelId, BuildingModelId, FloorModelId, HvacModelId);
+
+        // Call APIs to delete the models.
+        // Note that we are blocking the async API call. This is to ensure models are deleted in an order such that no other models are referencing it.
```

Comment:
We should have been able to schedule the async API on a single thread, and forced them to run sequentially; however, I didn't have success with that.
I am blocking the async API call for now, until I figure that out.